tests/api/route-services/work-response.test.js

```diff
@@ -124,35 +105,22 @@ describe('tests for the work-response service', () => {
         },
       ],
     };
+
     const w = new WorkResponseService(io, workResponse);
     const expectedResponse = JSON.parse(JSON.stringify(workResponse));
     delete expectedResponse.results[0].type;
 
-    w.handleResponse().then(() => {
-      expect(toSpy).toHaveBeenCalledTimes(1);
-      expect(toSpy).toHaveBeenCalledWith('mysocketid');
-      expect(emitSpy).toHaveBeenCalledTimes(1);
-      expect(emitSpy).toHaveBeenCalledWith('WorkResponse-55', expectedResponse);
-      return done();
+    client.on(`WorkResponse-${workResponse.request.uuid}`, (res) => {
+      expect(res).toEqual(expectedResponse);
+      done();
```

Comment:
Instead of the many calls to `emitSpy`, `toSpy`, etc., we can just directly check if the client got the message, and if it was the appropriate message to the appropriate channel, which is much more realistic.